src/Dev/Deprecation.php

```diff
@@ -41,6 +40,7 @@ class Deprecation
     const SCOPE_METHOD = 1;
     const SCOPE_CLASS = 2;
     const SCOPE_GLOBAL = 4;
+    const SCOPE_CONFIG = 8;
```

Comment:
> Called from [SomeClass->someMethod]

This doesn't make sense for config defined in yml.

We *could* add more fidelity here, though honestly it's pretty easy to use the stack traces as they are now. For calls to `Config::inst()->get()` or `Config::modify()->set()` the stack trace is on line 5, for example I commented out `$this->expectDeprecation();` in DeprecationTest::testConfigSet() and phpunit error + stack trace I get is

```
1) SilverStripe\Dev\Tests\DeprecationTest::testConfigSet
Config SilverStripe\Dev\Tests\DeprecationTest\DeprecationTestObject.first_config is deprecated. My first config message.

/var/www/vendor/silverstripe/framework/src/Dev/Deprecation.php:221
/var/www/vendor/silverstripe/config/src/Collections/MemoryConfigCollection.php:121
/var/www/vendor/silverstripe/config/src/Collections/DeltaConfigCollection.php:275
/var/www/vendor/silverstripe/config/src/Collections/DeltaConfigCollection.php:176
/var/www/vendor/silverstripe/framework/tests/php/Dev/DeprecationTest.php:115
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:97
/var/www/vendor/bin/phpunit:118
```

i.e. you can easily see it on line 5 of the stack trace - this will be trivial for projects because it will be the first non-vendor directory

I don't see any need to differentiate the messaging between `Config::inst()->get()` or `Config::modify()->set()`

If we try to add in the "called from" info, it gets a bit complicated because we need to:
a) it's extra work, and as noted above I think the stack trace if more than adequate
b) it gets very messy for yml config that throws errors on ?flush vs `Config::modify()->set()` - so again, more work

IMO this isn't worth it

Reply:
I still think it would be better if we had the "called from" portion in the deprecation message and I don't think those are good enough reasons to not include it - but I also don't think it's worth holding up this PR (which in turn is holding up other work) just for that. Ultimately if people find it would be useful to include that portion in the message I guess they can open a PR implementing it.